.github/files/gh-autorelease/files/autorelease.sh

```diff
@@ -48,76 +55,78 @@ echo "::group::Creating $SLUG.zip"
 git archive -v --output="$SLUG.zip" --prefix="$SLUG/" HEAD 2>&1
 echo "::endgroup::"
 
-## Create the release note.
-# Extract the changelog section.
-echo "::group::Extracting release notes"
-if [[ ! -f CHANGELOG.md ]]; then
-	echo '::endgroup::'
-	echo '::error::No CHANGELOG.md for release notes.'
-	exit 1
-fi
-SCRIPT="
-	/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
-		bc
+if [[ -z "$ROLLING_MODE" ]]; then
+	## Create the release note.
+	# Extract the changelog section.
+	echo "::group::Extracting release notes"
+	if [[ ! -f CHANGELOG.md ]]; then
+		echo '::endgroup::'
+		echo '::error::No CHANGELOG.md for release notes.'
+		exit 1
+	fi
+	SCRIPT="
+		/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
+			bc
+			:a
+			n
+			/^## / {
+				q
+			}
+			:c
+			s/^## \[([^]]+)\]/## \1/
+			p
+			ba
+		}
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
+	if [[ -z "$ENTRY" ]]; then
+		echo '::endgroup::'
+		echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
+		exit 1
+	fi
+
+	# Strip unwanted sections.
+	SCRIPT="
 		:a
-		n
-		/^## / {
-			q
+		/^### .* This section will not be copied to readme\.txt/ {
+			:b
+			n
+			/^#/ ba
+			bb
 		}
-		:c
-		s/^## \[([^]]+)\]/## \1/
 		p
-		ba
-	}
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
-if [[ -z "$ENTRY" ]]; then
-	echo '::endgroup::'
-	echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
-	exit 1
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
+
+	echo "Release notes:"
+	echo "-----"
+	echo "$ENTRY"
+	echo "-----"
+	echo "::endgroup::"
+else
+	## Using a brief explanation for the rolling release note.
+	ENTRY="### Rolling release based on the trunk branch."
 fi
 
-# Strip unwanted sections.
-SCRIPT="
-	:a
-	/^### .* This section will not be copied to readme\.txt/ {
-		:b
-		n
-		/^#/ ba
-		bb
-	}
-	p
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
-
-echo "Release notes:"
-echo "-----"
-echo "$ENTRY"
-echo "-----"
-echo "::endgroup::"
+if [[ "true" == "$ROLLING_MODE" ]]; then
+	echo "::group::Deleting stale rolling release"
+	gh release delete "$TAG" --cleanup-tag -y
+	echo "::endgroup::"
+fi
 
 echo "::group::Creating release"
-curl -v -L \
-	--write-out '%{response_code}' \
-	--output out.json \
-	--request POST \
-	--header "authorization: Bearer $API_TOKEN_GITHUB" \
-	--header 'content-type: application/json' \
-	--header 'accept: application/vnd.github.v3+json' \
-	--url "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/releases" \
-	--data "$(jq -n --arg tag "$TAG" --arg sha "$GITHUB_SHA" --arg title "$TITLE" --arg body "$ENTRY" '{ tag_name: $tag, target_commitish: $sha, name: $title, body: $body}')" \
-	2>&1 > code.txt
-cat out.json
-echo
-[[ "$(<code.txt)" =~ ^2[0-9][0-9]$ ]] || exit 1
+if ! gh release create "$TAG" \
+	--notes "$ENTRY" \
+	--target "$GITHUB_SHA" \
+	--title "$TITLE"; then
+	exit 1
+fi
```

Comment:
Thanks to the `-eo pipefail` at the top of the script, you don't need to explicitly check for failure.

Also, unless it breaks Atomic (we should ask them), we may as well throw in the `--prerelease` flag.
```suggestion
gh release create "$TAG" \
	--prerelease \
	--notes "$ENTRY" \
	--target "$GITHUB_SHA" \
	--title "$TITLE"
```

Reply:
Oh, wait.. We only want `--prerelease` if it's the rolling release. So that'll have to be more complicated. something like
```suggestion
ADDITIONAL_ARGS=()
if [[ -n "$ROLLING_RELEASE" ]]; then
	ADDITIONAL_ARGS+=( --prerelease )
fi
gh release create "$TAG" \
	"${ADDITIONAL_ARGS[@]}" \
	--notes "$ENTRY" \
	--target "$GITHUB_SHA" \
	--title "$TITLE"
```

Reply:
Noticed that suggestion only after I accepted the previous, so I added another commit, thanks!